### PR TITLE
MSR: A4 CM

### PR DESCRIPTION
**Pull request by Miepee**

### CODE REVIEW

File: randovania/games/samus_returns/logic_database/Area 4 - Crystal Mines.txt [hunk 1]
```diff
@@ -615,6 +666,11 @@ Extra - asset_id: collision_camera_009
   > Tunnel to Gamma Arena
       Morph Ball and Varia Suit
 
+> TODO: future dock; Heals? False
+  * Layers: default
+  > Door to Green Cystal Dugout
```
Comment (dyceron) [inline randovania/games/samus_returns/logic_database/Area 4 - Crystal Mines.txt:671]: I just noticed this room is spelled wrong lol. Should be `Green Crystal Dugout` 😛

File: randovania/games/samus_returns/logic_database/Area 4 - Crystal Mines.txt [hunk 2]
```diff
@@ -708,10 +772,15 @@ Extra - asset_id: collision_camera_012
...
-          Gravity Suit and Simple IBJ
           All of the following:
-              Infinite Bomb Jump (Intermediate) and Lay Bomb
-              Spring Ball or Mid-Air Morph (Intermediate)
+              # IBJ from water
+              Lay Bomb
+              # With gravity
+              Grapple Beam and Simple IBJ
+              # Without gravity
+              Infinite Bomb Jump (Intermediate)
```
Comment (dyceron) [inline randovania/games/samus_returns/logic_database/Area 4 - Crystal Mines.txt:781]: This grouping is wrong and this says Grapple and not Gravity

File: randovania/games/samus_returns/logic_database/Area 4 - Crystal Mines.txt [hunk 3]
```diff
@@ -725,11 +794,17 @@ Extra - asset_id: collision_camera_012
...
-              Infinite Bomb Jump (Intermediate) and Lay Bomb
-              Spring Ball or Mid-Air Morph (Intermediate)
+              # IBJ from water
+              Lay Bomb
+              # With Gravity
+              Gravity Suit and Simple IBJ
+              # Without Gravity
+              Infinite Bomb Jump (Intermediate)
+          # IBJ from outside of water
+          Mid-Air Morph (Intermediate) and Infinite Bomb Jump (Beginner) and Lay Bomb
```
Comment (dyceron) [inline randovania/games/samus_returns/logic_database/Area 4 - Crystal Mines.txt:807]: Same as above comment

File: randovania/games/samus_returns/logic_database/Area 4 - Crystal Mines.txt [hunk 4]
```diff
@@ -1103,8 +1190,15 @@ Extra - asset_id: collision_camera_AfterChase_001
...
+      Trivial
+
+> Next to Middle item; Heals? False
+  * Layers: default
+  > Pickup (Missile Tank Middle)
       Any of the following:
-          # Roll under or defeat Paraby swarm
-          Defeat Swarms
-          Morph Ball and Dodge Enemies (Intermediate)
+          Lay Bomb
```
Comment (dyceron) [inline randovania/games/samus_returns/logic_database/Area 4 - Crystal Mines.txt:1199]: Lay Power Bomb

File: randovania/games/samus_returns/logic_database/Area 4 - Crystal Mines.txt
Comment (dyceron) [inline randovania/games/samus_returns/logic_database/Area 4 - Crystal Mines.txt:1]: Add TODO somewhere in `Diggernaut Excavation Tunnels` to simplify and remove extra nodes

File: randovania/games/samus_returns/logic_database/Area 4 - Crystal Mines.txt (hunk 1 above)
Comment (Miepee) [reply randovania/games/samus_returns/logic_database/Area 4 - Crystal Mines.txt:671]: out of scope for this PR :P

File: randovania/games/samus_returns/logic_database/Area 4 - Crystal Mines.txt [hunk 5]
```diff
@@ -112,9 +117,14 @@ Extra - asset_id: collision_camera_001
   * Extra - actor_type: doorpowerpower
   > Above Aeion Refill
       Any of the following:
-          High Jump Boots or Morph Ball or Space Jump or Wall Jump (Beginner)
```
[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/games/samus_returns/logic_database/Area 4 - Crystal Mines.txt:115]: disagree with readability here. Having the template with the appropriate comments makes this more readable and maintainable

**Review by dyceron — APPROVED** [review]

**Pull request merged**